python/tests/integration/arcticdb/test_read_batch_more.py

```diff
@@ -0,0 +1,364 @@
+    """
+        Concatenates and sorts row range indexed dataframes
+    """
+    result = pd.concat(df_args[0:2], copy=True)
+    for idx in range(2,len(df_args),1):
+        result = pd.concat([result, df_args[idx]],copy=True)
+    result.sort_index(inplace=True) # We need to sort it at the end
+    return result
+
+def test_read_batch_2tables_7reads_different_slices(arctic_library):
+    """
+        Test aims to check if combined read of couple of DF, with several 
+        reads from each, which filters different subsections of the timeframes
+        is correct, in other words each read request is isolated from each other.
+        Covers columns, as_of and date_range parameters of read_batch() function
+    """
+    lib = arctic_library
+
+    symbol1 = "sym1"
+    df1_0 = create_df_index_datetime(7, 0, 5)
+    df1_1 = create_df_index_datetime(7, 4, 6)
+    df1_2 = create_df_index_datetime(7, 6, 10)
+    df1_3 = create_df_index_datetime(7, 0, 10)
+    df1_till2 = dataframe_simulate_arcticdb_update_static(df1_0, df1_1)  # DF of state 0+1
+    df1_till3 = dataframe_concat_sort(df1_till2, df1_2) # DF of state 0+1+2
+    df1_all = dataframe_simulate_arcticdb_update_static(df1_till3, df1_3)
+
+    symbol2 = "sym2"
+    df2_0 = create_df_index_datetime(200, 0, 100)  
+    df2_1 = create_df_index_datetime(200, 100, 200)
+    df2_2 = create_df_index_datetime(200, 200, 300)
+    df2_till2 = dataframe_concat_sort(df2_0, df2_1) # DF of state 0+1
+    df2_all = dataframe_concat_sort(df2_till2, df2_2)
+    # A DF with certain colums selected
+    columns_to_select = ["COL_1", "COL_33", "COL_155"]
+    df2_all_col_filtered = df2_all.loc[:,columns_to_select] 
+    # Here we would like to produce a DF without several first and last rows
+    start = df2_all.index[4]
+    end = df2_all.index[-5]
+    date_range = (start, end)
+    df2_all_without_first_and_last = dataframe_filter_with_datetime_index(df2_all, start, end)
+    # Here we would like to produce a DF without several first and last rows
+    # and only two colums one of the first and one of the last
+    columns_to_select1= ["COL_1", "COL_198"]
+    start1 = df2_0.index[1]
+    end1 = df2_0.index[-2]
+    date_range1 = (start1, end1)
+    tmp = df2_0.loc[:,columns_to_select1]
+    df2_0_allfilters = dataframe_filter_with_datetime_index(tmp, start1, end1)
+
+    symbol3 = "sym3" # non-existing
+
+    lib.write(symbol1, df1_0)
+    lib.update(symbol1, df1_1)
+    lib.append(symbol1, df1_2)
+    lib.update(symbol1, df1_3)
+
+    lib.write(symbol2, df2_0)
+    lib.append(symbol2, df2_1)
+    lib.append(symbol2, df2_2)
+
+    # Check Pandas update logic (simulating arctic append/update operations)
+    assert_frame_equal(df1_all, df1_3)
+    
+    # Assure last version is exactly what we expect
+    symbol1_data_sorted = lib.read(symbol1).data
+    assert_frame_equal(df1_all, symbol1_data_sorted)
+    assert_frame_equal(df1_3, symbol1_data_sorted)
```

Comment:
Given df1_all is equal to df1_3 what's the point of having two assertions, equals is transitive relation

Reply:
agree